homeassistant/components/cloud/iot.py

```diff
@@ -228,7 +228,10 @@ def async_handle_message(hass, cloud, handler_name, payload):
 def async_handle_alexa(hass, cloud, payload):
     """Handle an incoming IoT message for Alexa."""
     if not cloud.alexa_enabled:
-        return alexa.turned_off_response(payload)
+        # pylint: disable=protected-access
+        directive = alexa._AlexaDirective(payload)
```

Comment:
Instead to access a protected function (look like a design error), can we make a public function or helper function for it?

Reply:
I couldn't think of a use case for doing this outside of this one thing. Seems to me like a reasonable thing to do, as this particular function has a legitimate reason for needing to reach into the low-level details of the Alexa API. More of an exception than the norm.

Reply:
Alternately, maybe it's better to respond to this request in a way that doesn't require knowledge of the alexa API? Then we avoid any coupling entirely. For example, maybe just an HTTP 404 with a simple text body which links to instructions for enabling alexa cloud or something. I don't use cloud so I don't really know if that would be any different from a UX perspective.

Reply:
this is being routed directly back to Alexa, so it needs to be in the Alexa format.

Reply:
OK, how about we just pass an `enabled` parameter to `async_handle_message`. Then smart_home.py doesn't need to expose any low-level details, nor does iot.py need to care.